packit/config/package_config.py

```diff
@@ -81,18 +79,69 @@ def get_from_dict(
             raw_dict.setdefault("upstream_package_name", repo_name)
             raw_dict.setdefault("downstream_package_name", repo_name)
 
+    @classmethod
+    def get_from_dict(
+        cls,
+        raw_dict: dict,
+        config_file_path: Optional[str] = None,
+        repo_name: Optional[str] = None,
+        search_specfile: Optional[Callable[..., Optional[str]]] = None,
+        **specfile_search_args,
+    ) -> "PackageConfig":
+        # required to avoid cyclical imports
+        from packit.schema import PackageConfigSchema
+
+        # we need to process defaults first so they get propagated to JobConfigs
+        raw_dict.setdefault("config_file_path", config_file_path)
+
+        monorepo = False
+        if packages := raw_dict.get("packages", None):
+            monorepo = True
+            for package in packages.values():
+                up_url_key = "upstream_project_url"
+                if up_url_key in raw_dict:
+                    # propagate it to any monorepo sub-package
+                    package[up_url_key] = raw_dict[up_url_key]
+                cls.set_defaults(
+                    package, repo_name, search_specfile, **specfile_search_args
+                )
+        else:
+            cls.set_defaults(
+                raw_dict, repo_name, search_specfile, **specfile_search_args
+            )
+
         package_config = PackageConfigSchema().load(raw_dict)
 
+        for name, package in package_config.packages.items():
+            # inject is_sub_package info,
+            # only package_config knows it
+            package.is_sub_package = bool(monorepo)
```

Comment:
Why we are changing the type here?

Since we don't use it anywhere else, can't we use
```suggestion
            package.is_sub_package = "packages" in raw_dict
```

And remove the line `97` and `99`?

Or maybe use the following to be really explicit what it means:

```python
is_monorepo = "packages" in raw_dict
```

Reply:
> Why we are changing the type here?

I don't know 😅 I was tired the linter was suggestion me this and I blindly followed it 🙈 good catch!

I like this one `is_monorepo = "packages" in raw_dict` I think I will use it.